pkg/types/llo/types.go

```diff
@@ -95,14 +191,82 @@ type Transmitter ocr3types.ContractTransmitter[ReportInfo]
 // address, i.e. the only address allowed to verify reports from this channel
 // https://smartcontract-it.atlassian.net/browse/MERC-3492
 type ChannelDefinition struct {
-	ReportFormat ReportFormat
-	// Specifies the chain on which this channel can be verified. Currently uses
-	// CCIP chain selectors.
-	ChainSelector uint64
-	// We assume that StreamIDs is always non-empty and that the 0-th stream
-	// contains the verification price in LINK and the 1-st stream contains the
-	// verification price in the native coin.
-	StreamIDs []StreamID
+	ReportFormat ReportFormat `json:"reportFormat"`
+	// StreamIDs contains a list of streams to be observed for this channel.
+	StreamIDs []StreamID `json:"streamIDs"`
+	// Aggregators contains a list of aggregator methods used for each stream
+	// e.g. "median", "mode" or other more exotic methods.
+	// StreamIDs and Aggregators must have the same length.
```

Comment:
is there any limitation to why StreamIDs and Aggregators are split up? If they are a 1:1 pair wouldn't it just be easier + less error prone to define them together?
```
Things []Thing json:"things"`

type Thing struct {
   s StringId
   a Aggregator 
}

```

Reply:
Original reason is that it maps more closely to how you'd represent that data on-chain, and it keeps all data types that may need to be interpreted on-chain simple.

Reply:
It's a fair point though, it might be safer to use a tuple/composite type here.